SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingServiceTests.cs

```diff
@@ -0,0 +1,148 @@
+                await groupService.AddUserToGroup(group1.Id, user4.Id);
+
+                await groupService.AddUserToGroup(group2.Id, user1.Id);
+                await groupService.AddUserToGroup(group2.Id, user2.Id);
+                await groupService.AddUserToGroup(group2.Id, user3.Id);
+                await groupService.AddUserToGroup(group2.Id, user4.Id);
+            }
+        }
+
+        [TestMethod]
+        [DataRow(-1)]
+        [DataRow(0)]
+        public async Task GeneratPairings_RequiresPositiveGroupId(int groupId)
+        {
+            using (ApplicationDbContext context = new ApplicationDbContext(Options))
+            {
+                PairingService pairingService = new PairingService(context);
+                await Assert.ThrowsExceptionAsync<ArgumentOutOfRangeException>(async () => 
+                    await pairingService.GeneratePairings(groupId));
+            }
+        }
+
+
+        [TestMethod]
+        public async Task GeneratePairings_GeneratesUniquePairings()
+        {
+            using (ApplicationDbContext context = new ApplicationDbContext(Options))
+            {
+                PairingService pairingService = new PairingService(context);
+                List<Pairing> pairings = await pairingService.GeneratePairings(1);
+                
+                var santaIds = pairings.Select(x => x.SantaId).ToList();
+                var recipientIds = pairings.Select(x => x.RecipientId).ToList();
+
+                Assert.AreEqual(4, santaIds.Distinct().Count());
+                Assert.AreEqual(4, recipientIds.Distinct().Count());
+            }
+        }
+
+        [TestMethod]
+        public async Task GeneratePairings_ContainNoSelfGifters()
+        {
+            using (ApplicationDbContext context = new ApplicationDbContext(Options))
+            {
+                PairingService pairingService = new PairingService(context);
+                List<Pairing> pairings = await pairingService.GeneratePairings(1);
+
+                foreach (var p in pairings)
+                {
+                    Assert.AreNotEqual<int>(p.SantaId, p.RecipientId);
+                }
+            }
+        }
+
+        [TestMethod]
+        public async Task GeneratePairings_AddPairingsThatDifferOnlyByGroup()
+        {
+            using (ApplicationDbContext context = new ApplicationDbContext(Options))
+            {
+                PairingService pairingService = new PairingService(context);
+                List<Pairing> pairings1 = await pairingService.GeneratePairings(1);
+                List<Pairing> pairings2 = await pairingService.GeneratePairings(2);
+
+                List<int> pairing1UserIds = pairings1.Select(x => x.SantaId).ToList();
+                List<int> pairing2UserIds = pairings2.Select(x => x.SantaId).ToList();
+
+                Assert.AreEqual(pairings1.Select(x => x.OriginGroupId).First(), 1);
+                Assert.AreEqual(pairings2.Select(x => x.OriginGroupId).First(), 2);
```

Comment:
In these cases you probably want to verify that all object in the collection have the group id, rather than just the first.
```suggestion
                Assert.IsTrue(pairings2.All(x => x.OriginalGroupId == 2));
```